strings/base_com_ptr.h

```diff
@@ -59,33 +59,31 @@ WINRT_EXPORT namespace winrt
 
         com_ptr& operator=(com_ptr const& other) noexcept
         {
-            copy_ref(other.m_ptr);
+            com_ptr cpy(other);
+            swap(*this, cpy);
             return*this;
         }
 
         com_ptr& operator=(com_ptr&& other) noexcept
```

Comment:
Take by value to fold both move and copy into a single operator?

Reply:
I thought about it but decided against it because it is easier for me to understand. If I have only one assignment operator, the reader will surely ask where the move operator is or whether this was perhaps forgotten.
Furthermore, it has been done in the implementation of the STL shared_ptr (Microsoft, libstd++, ...) as well and I didn't want to put myself in danger of using another implementation in such a critical place.

Reply:
A small comment can avoid the concerns about thinking that move was forgotten

Reply:
I will change it if another one is for it.

Reply:
Sorry, the situation has changed so move and copy are different now. @roxk exception safety should not be a problem here, because the functions are not allowed to throw (noexcept) but the current implementation fulfills that as well.